pybio/spec/fields.py

```diff
@@ -22,9 +23,58 @@ def resolve_local_path(path_str: str, context: dict) -> pathlib.Path:
         return local_path.resolve()
 
 
+def resolve_doi(uri: ParseResult) -> ParseResult:
+    if uri.scheme.lower() != "doi" and not uri.netloc.endswith("doi.org"):
+        return uri
+
+    doi = uri.path.strip("/")
+
+    url = "https://doi.org/api/handles/" + doi
+    r = json.loads(requests.get(url).text)
+    response_code = r["responseCode"]
+    if response_code != 1:
+        raise RuntimeError(f"Could not resolve doi {doi} (responseCode={response_code})")
```

Comment:
It's nice for the consumer of this library if we create library specific exceptions and raise as needed. Create a library specific exception that inherits `Exception` and make further child exceptions from that parent as needed.

Reply:
good idea, I started this in https://github.com/bioimage-io/python-bioimage-io/pull/2/commits/0d0626552f699b77c858ec6d9e9f62ea3304857c